### ISSUE TITLE
Consistent crashes for v0.16.0rc1

### ISSUE THREAD

**Issue by Sjors** [bitcoin/bitcoin#12328]
Possibly the same as #12323, but although I'm seeing the file descriptor errors, the crashes don't immediately follow that.

1. Crash log: http://termbin.com/h7jy

```
System error: CAutoFile::write: write failed: unspecified iostream_category error"
```

2. crash log (16bac24f6): http://termbin.com/b3250

```
2018-02-01 17:45:37 trying connection seed.testnet.bitcoin.sprovoost.nl lastseen=0.0hrs
2018-02-01 17:45:37 ERROR: WriteBlockToDisk: ftell failed
2018-02-01 17:45:37 *** Failed to write block
2018-02-01 17:45:37 Error: Error: A fatal internal error occurred, see debug.log for details
2018-02-01 17:45:37 ERROR: ProcessNewBlock: AcceptBlock FAILED ()
```

3. crash log (eadb2dacc): http://termbin.com/f3f6 

```
2018-02-01 17:56:44 ERROR: UndoWriteToDisk: ftell failed
2018-02-01 17:56:44 *** Failed to write undo data
2018-02-01 17:56:44 Error: Error: A fatal internal error occurred, see debug.log for details
```

4. crash log (126000ba9): http://termbin.com/fh3e

```
2018-02-01 18:03:38 trying connection seed.testnet.bitcoin.sprovoost.nl lastseen=0.0hrs
2018-02-01 18:03:38 *** System error: CAutoFile::write: write failed: unspecified iostream_category error
2018-02-01 18:03:38 Error: Error: A fatal internal error occurred, see debug.log for details
2018-02-01 18:03:38 ERROR: ProcessNewBlock: AcceptBlock FAILED ()
```

5. crash log (6e89de5ba): http://termbin.com/5jdf

```
2018-02-01 18:22:27 trying connection seed.testnet.bitcoin.sprovoost.nl lastseen=0.0hrs
2018-02-01 18:22:28 ERROR: UndoWriteToDisk: ftell failed
2018-02-01 18:22:28 *** Failed to write undo data
``` 

6. crash log (2b839abd3): http://termbin.com/9onp

```
018-02-01 18:58:07 trying connection seed.testnet.bitcoin.sprovoost.nl lastseen=0.0hrs
2018-02-01 18:58:07 ERROR: UndoWriteToDisk: ftell failed
2018-02-01 18:58:07 *** Failed to write undo data
```

7. crash log (62e764219): http://termbin.com/ruhy

```
2018-02-01 19:16:46 received block 00000000027ec99e54ff872875741479dd64dad1847cee9b6d03233fb5ec2e6c peer=3
2018-02-01 19:16:46 trying connection seed.testnet.bitcoin.sprovoost.nl lastseen=0.0hrs
2018-02-01 19:16:46 *** System error: CAutoFile::write: write failed: unspecified iostream_category error
2018-02-01 19:16:46 Error: Error: A fatal internal error occurred, see debug.log for details
2018-02-01 19:16:46 ERROR: ProcessNewBlock: AcceptBlock FAILED ()
```

After the error it usually happily processes a few more blocks before shutting down.

When running QT I saw a different error message: `ERROR: WriteBlockToDisk: ftell failed`

So far I've only seen this problem on testnet, but I haven't tried an IDB on mainnet.

bitcoin.conf:

```
testnet=1
server=1
rpcuser=...
rpcpassword=...
listen=0
```

Crash happens without `server=1` and rpc stuff as well.

I'm currently running a `git bisect` between `v0.15.1` and `v0.16.0rc1`.

```sh
./configure --disable-tests --disable-bench --with-miniupnpc=no
make src/bitcoind

src/bitcoind -debug=1 --daemon
tail -f [...]testnet3/debug.log
```

I stopped bitcoind and deleted the datadir after each run. I consider a run successful if it doesn't crash within 25 minutes.

See also todays IRC [discussion](https://botbot.me/freenode/bitcoin-core-dev/2018-02-01/?msg=96378789&page=2).

**Comment by Sjors** [comment]
Git bisect blames 62e764219 (#4440710), which matches the suspected culprit mentioned in #12329.

I also tried the backported #12326 (e54c1ac): no crash (at least for 20 minutes)

I commented out my DNS seed and retried v0.16.0rc1: no crash (at least for 22 minutes)

**Comment by MrTuoi** [comment]
I do not think it collapsed any minute does not matter, if we make the restart process faster than the process of collapse is completely kept loop continuous.

**Comment by laanwj** [comment]
Thanks for trying to bisect this. 

It should be fixed by #12326 + #12329.
